### PR TITLE
refactor logprob processor layer

### PR DESCRIPTION
**Pull request by Qiaolin-Yu**
## Summary

- Introduce `python/sglang/srt/layers/logprob_processor.py` as the dedicated home for logprob processing, starting with the input (prefill) side.
- The branch was reset to current `main` and the extraction redone on today's code (the original diff had drifted ~4.8k commits behind); please review the PR diff rather than the intermediate commits.

## Changes

- Move `layers/utils/logprob.py` -> `layers/logprob_processor.py` (pure rename, 4 importers updated).
- Extract `InputLogprobProcessor` from `LogitsProcessor`: chunk/non-chunk gating, `process_input_logprobs`, and `process_input_logprobs_by_chunk` move onto the new class unchanged; `LogitsProcessor.forward` delegates to it.
- lm_head compute is injected as a `get_logits_fn` callable so the class stays decoupled from the TP-gather / logits-buffer machinery; the DP-attention skip flag is passed in as an argument.

## Behavior

- No behavior change intended: env reads keep the same lifetime, gating is computed per forward as before, and all moved bodies are byte-identical modulo two declared renames (`self._get_logits` -> `get_logits_fn`, `self.do_tensor_parallel_all_gather_dp_attn` -> `skip_chunking_for_dp_attn`).
- Only visible surface change is the module path (`layers.utils.logprob` -> `layers.logprob_processor`); no in-tree stragglers remain.

## Mechanical verification

Scripts and raw output: https://gist.github.com/hnyls2002/1266e3fd3e1dcde4ec68e301add02026

- AST equivalence of both moved methods under the declared rename rules: 2/2 OK.
- Full-diff hunk accounting vs `main`: 9/9 OK — peripheral files assert import-only / docstring-only changes; the two core files pass a bidirectional line-multiset check where each side's surplus must exactly match the declared scaffolding (class/signature/delegation lines).
- CI on this branch: `test_srt_endpoint.py`, `test_lora_hf_sgl_logprob_diff.py` (chunked path incl. lm_head multi-pass), `test_spec_ngram.py`, and `test_original_logprobs.py` all pass.

## Follow-up

- `OutputLogprobProcessor` (sampler side) and a unified logprob result dataclass, as separate commits.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #29620073711](https://github.com/sgl-project/sglang/actions/runs/29620073711)
Latest PR Test (Extra): :x: [Run #29620072939](https://github.com/sgl-project/sglang/actions/runs/29620072939)

### CODE REVIEW

**Comment by hnyls2002** [comment]
/rerun-test registered/core/test_srt_endpoint.py registered/lora/test_lora_hf_sgl_logprob_diff.py registered/spec/test_spec_ngram.py registered/sampling/test_original_logprobs.py
